### PR TITLE
Scene Tree Dock: added ability to deselect items when clicking on empty space

### PR DESCRIPTION
**Pull request by Krakean**
This is how it looks in action: https://i.imgur.com/tMECINu.gifv

### CODE REVIEW

**Comment by reduz** [comment]
does not seem vital for 3.0, wonder what others think of this feature.

**Comment by Krakean** [comment]
@reduz its just an slight cosmetic adjustment, should not introduce any issues so in my opinion pretty safe to merge into 3.0 :)

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -1923,6 +1923,9 @@ int Tree::propagate_mouse_event(const Point2i &p_pos, int x_ofs, int y_ofs, bool
 				c = c->next;
 				item_h += child_h;
 			}
+
+			if (!c && !p_mod->get_shift() && !p_mod->get_control() && !p_mod->get_command() && !click_handled && p_button != BUTTON_RIGHT)
```
Comment (letheed) [inline scene/gui/tree.cpp]: Should this work with BUTTON_MIDDLE?

Comment (Krakean) [reply scene/gui/tree.cpp]: Yep. I just ignore right mouse button in this case because it has popup menu

Comment (letheed) [reply scene/gui/tree.cpp]: But selection does not work with the MMB, so I don't see why it should with deselection

Comment (Krakean) [reply scene/gui/tree.cpp]: Well, then it will not deselect. So nothing broken after all, right? :)

**Pull request merged**